benchmark/CMakeLists.txt

```diff
@@ -15,6 +15,18 @@ if(CMAKE_BUILD_TYPE MATCHES "Release|RelWithDebInfo")
     endif()
 endif()
 
+# Macro to set properties for Xcode targets
+macro(set_xcode_properties TARGET_NAME)
+    if(CMAKE_GENERATOR STREQUAL Xcode)
+        set_target_properties(${TARGET_NAME} PROPERTIES
+            XCODE_ATTRIBUTE_ENABLE_AVX YES
+            XCODE_ATTRIBUTE_ENABLE_AVX2 YES
+            XCODE_ATTRIBUTE_OTHER_CPLUSPLUSFLAGS "-mavx -mavx2"
```

Comment:
will run this on my M1 machine later on today, but i think you'll need another layer here of 
```cmake
if (${TARGET_PLATFORM} == X86_64) 
    // enable avx
endif()
```
will find the code i've used before which will do this only when AVX is actually detected on the machine, but i think this will need to be updated to work on ARM

Reply:
Wait, doesn't XCode in M1 allow you to cross compile transparently to x86-64?

Reply:
I think it would be cool to try and keep the code (where possible) architecture independent, otherwise could become a pain to try and go back and wrangle it to work in the future.

Reply:
Keeping code as arch independent as possible seems critical if our objectives are to work everywhere. Which I think they are.

Reply:
Yes, friends, architecture independence is a primary objective.  But we have the freedom to draft and experiment architecture specific.